### PR TITLE
Add Schedule & Engine - Pipeline level config to top panel

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds Schedule & Engine to top panel for pipeline configuration in Hydrator++ studio view.

Bamboo build: http://builds.cask.co/browse/CDAP-DRC3941-3
##### Note:
- Need to add the top panel view to published pipeline. It is not yet there.

### CODE REVIEW

File: cdap-ui/app/directives/my-card/my-card-crtl.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+angular.module(PKG.name + '.commons')
+  .controller('MyCardCtrl', function($scope) {
+    this.minimize = false;
+    this.title = $scope.title;
```
Comment (elfenheart) [inline cdap-ui/app/directives/my-card/my-card-crtl.js]: you can set bindToController to true at the directive level, so you don't have to do this

File: cdap-ui/app/features/hydratorplusplus/toppanel.less [hunk 2]
```diff
@@ -107,17 +107,30 @@ body.theme-cdap {
         padding-left: 10px;
       }
     }
+
+    .pipeline-settings {
+      background: lightgray;
+      position: fixed;
+      top: 93px;
+      right: 0;
+      z-index: 100001;
```
Comment (elfenheart) [inline cdap-ui/app/features/hydratorplusplus/toppanel.less]: might need to reduce this z-index to be behind the cdap loading modal

**Comment by elfenheart** [comment]
Couple of questions: 
1. Complex Cron Editor
2. ETL Realtime view not yet there

File: cdap-ui/app/directives/my-pipeline-settings/my-batch-pipeline-settings/my-batch-pipeline-settings-ctrl.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  onEngineChange() {
+    this.actionCreator.setEngine(this.engine);
+  }
+  changeScheduler () {
+    this.initialCron = this.store.getDefaultSchedule();
+    this.cron = this.initialCron;
+  }
+}
+
+MyBatchPipelineSettingsCtrl.$inject = ['GLOBALS', '$scope'];
```
Comment (elfenheart) [inline cdap-ui/app/directives/my-pipeline-settings/my-batch-pipeline-settings/my-batch-pipeline-settings-ctrl.js:57]: no need to do this now. Just put `'ngInject';` right after constructor

Comment (ajainarayanan) [inline cdap-ui/app/directives/my-pipeline-settings/my-batch-pipeline-settings/my-batch-pipeline-settings-ctrl.js:57]: Yeah `ngAnnotate` was throwing some error or was not annotating es6 classes for some reason.

Comment (elfenheart) [inline cdap-ui/app/directives/my-pipeline-settings/my-batch-pipeline-settings/my-batch-pipeline-settings-ctrl.js:57]: might not have the latest change from develop. No worries

**Comment by elfenheart** [comment]
👍  LGTM

File: cdap-ui/app/directives/my-card/my-card-crtl.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+angular.module(PKG.name + '.commons')
+  .controller('MyCardCtrl', function() {
+    this.minimize = false;
+    this.dominimize = function() {
```
Comment (anwar6953) [inline cdap-ui/app/directives/my-card/my-card-crtl.js:20]: should capitalize the m?

**Pull request merged**